include/library/spdm_requester_lib.h

```diff
@@ -141,6 +141,62 @@ libspdm_return_t libspdm_get_certificate_ex(void *spdm_context,
                                             void *cert_chain,
                                             const void **trust_anchor,
                                             size_t *trust_anchor_size);
+
+/**
+ * This function sends GET_CERTIFICATE to get certificate chain in one slot from the device,
+ * and allows the Integrator to specify the size of the certificate chain blocks.
+ *
+ * This function verify the integrity of the certificate chain.
+ * root_hash -> Root certificate -> Intermediate certificate -> Leaf certificate.
+ *
+ * If the peer root certificate hash is deployed,
+ * this function also verifies the digest with the root hash in the certificate chain.
+ *
+ * @param  spdm_context     A pointer to the SPDM context.
+ * @param  session_id       Indicates if it is a secured message protected via SPDM session.
+ *                          If session_id is NULL, it is a normal message.
+ * @param  slot_id          The number of slot for the certificate chain.
+ * @param  length           The length of the certificate chain block to retrieve.
+ * @param  cert_chain_size  On input, indicate the size in bytes of the destination buffer to store the digest buffer.
+ *                          On output, indicate the size in bytes of the certificate chain.
+ * @param  cert_chain       A pointer to a destination buffer to store the certificate chain.
+ **/
+libspdm_return_t libspdm_get_certificate_choose_length(void *spdm_context,
+                                                       const uint32_t *session_id,
+                                                       uint8_t slot_id,
+                                                       uint16_t length,
```

Comment:
Can we use `uint32_t`, this is for new SPDM spec?

Reply:
So if
- negotiated version is <= 1.3 then `length` must must be `<= 65535`
- negotiated version is >= 1.4 then libspdm uses `LargeOffset` and `LargeLength`.
?

Reply:
Please just check 0xFFFF in this version.

We can add more check for 1.4 later.

Reply:
For versions <= 1.3, even though we use uint32_t here, we will need to typecast it to uint16_t before passing it to libspdm_try_get_certificate in the function, after checking that length <= 0xFFFF. Trying to understand the motivation behind changing this to uint32 for the current version?